Formula/nerd-fonts-patcher.rb

```diff
@@ -0,0 +1,28 @@
+class NerdFontsPatcher < Formula
+  desc "Patch your own font"
+  homepage "https://github.com/ryanoasis/nerd-fonts#font-patcher"
+  url "https://github.com/ryanoasis/nerd-fonts/trunk/src", using: :svn
+  version = version "2.1.0"
+  license "MIT"
+
+  depends_on "fontforge"
+
+  resource "script" do
+    url "https://raw.githubusercontent.com/ryanoasis/nerd-fonts/v#{version}/font-patcher"
+    sha256 "3377615be4271f8bdeef66e6f2f82ac3f3cfb7b5677abe7b8e189409da048859"
+  end
+
+  def install
+    resource("script").stage { prefix.install "font-patcher" }
+    mkdir "src"
+    mv Dir["*"] - ["src"], "src"
+    prefix.install "src"
+    bin.write_exec_script prefix/"font-patcher"
+    inreplace bin/"font-patcher", "exec", "exec fontforge -script"
+  end
+
+  test do
+    stdout = shell_output bin/"font-patcher --version"
+    assert_match "Nerd Fonts: font-patcher (#{version})", stdout
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev Fixed.